common/src/main/java/com/wynntils/features/chat/InfoMessageFilterFeature.java

```diff
@@ -12,108 +12,100 @@
     public final Config<Boolean> hideSystemInfo = new Config<>(true);
 
     @RegisterConfig
     public final Config<Boolean> hideLevelUp = new Config<>(false);
 
-    @SubscribeEvent
-    public void onInfoMessage(ChatMessageReceivedEvent e) {
-        if (e.getRecipientType() != RecipientType.INFO) return;
+    @RegisterConfig
+    public final Config<Boolean> hidePartyFinder = new Config<>(false);
 
+    @SubscribeEvent
+    public void onMessage(ChatMessageReceivedEvent e) {
         CodedString msg = e.getOriginalCodedString();
         MessageType messageType = e.getMessageType();
 
-        if (messageType == MessageType.FOREGROUND) {
-            if (hideSystemInfo.get()) {
-                if (msg.getMatcher(SYSTEM_INFO).find()) {
-                    e.setCanceled(true);
-                    return;
-                }
-            }
+        if (hideWelcome.get()) {
+            e.setCanceled(processFilter(msg, messageType, WELCOME));
+            return;
+        }
 
-            if (hideWelcome.get()) {
-                if (msg.getMatcher(WELCOME_1).find()
-                        || msg.getMatcher(WELCOME_2).find()) {
-                    e.setCanceled(true);
-                    return;
-                }
-            }
-            if (hideLevelUp.get()) {
-                if (msg.getMatcher(LEVEL_UP_1).find()
-                        || msg.getMatcher(LEVEL_UP_2).find()) {
-                    e.setCanceled(true);
-                    return;
-                }
-            }
+        if (hideSystemInfo.get()) {
+            e.setCanceled(processFilter(msg, messageType, SYSTEM_INFO));
+            return;
+        }
 
-            if (hideWelcome.get()) {
-                if (msg.getMatcher(PRE_WELCOME_1).find()
-                        || msg.getMatcher(PRE_WELCOME_2).find()
-                        || msg.getMatcher(PRE_WELCOME_3).find()) {
-                    e.setCanceled(true);
-                    return;
-                }
-            }
-        } else if (messageType == MessageType.BACKGROUND) {
-            if (hideSystemInfo.get()) {
-                if (msg.getMatcher(BACKGROUND_SYSTEM_INFO).find()) {
-                    e.setCanceled(true);
-                    return;
-                }
-            }
+        if (hideLevelUp.get()) {
+            e.setCanceled(processFilter(msg, messageType, LEVEL_UP));
+            return;
+        }
 
-            if (hideLevelUp.get()) {
-                if (msg.getMatcher(BACKGROUND_LEVEL_UP_1).find()
-                        || msg.getMatcher(BACKGROUND_LEVEL_UP_2).find()) {
-                    e.setCanceled(true);
-                    return;
-                }
-            }
+        if (hidePartyFinder.get()) {
+            // Matches until the username in the party finder message
+            e.setCanceled(msg.startsWith(ChatFormatting.DARK_PURPLE + "Party Finder:" + ChatFormatting.RESET
+                    + ChatFormatting.LIGHT_PURPLE + " Hey "));
```

Comment:
Actually I didnt notice this before but why is this not just a regular regex?

Reply:
Do we need it to be a regular regex? Imo this is strict enough to only detect the pfinder messages and is faster than a normal regex

Reply:
When you have a feature containing 10+ regexes it is weird for a random one not to be a regex, there is no point in doing it. As for performance, it basically doesn't change a thing, you will never notice this.